### DISCUSSION TITLE
Simultaneous usage of app dir and pages dir

### DISCUSSION THREAD

**Discussion by yusifsalam** [vinissimus/next-translate#1043] [Q&A] [answered]
Hi,

I have an older project that was using the pages dir with next-translate and now that the app router is declared stable by the NextJS team, I thought I would start gradually migrating parts of the project to using the new app dir. 
So my question is: is it possible to use next-translate for both app dir and pages dir at the same time in order to allow for gradual migration? 

I tried creating a new test page inside the app dir and loading the translations there but the translation hook isn't able to load the key correctly - it just displays the key. At the same time, pages under the old pages dir continue working normally, meaning translation load correctly. 

Thank you!

**Comment by aralroca** [comment, marked as answer]
@yusifsalam Currently, the [next-translate-plugin](https://github.com/aralroca/next-translate-plugin) is taking account or `pages` or `app`, but not both at the same time. However, I understand can be necessary in a lot of cases to do this migration, so we can take a look in order to support both approaches. Feel free to contribute in [next-translate-plugin](https://github.com/aralroca/next-translate-plugin) in order to support it.
